Redesigned data entry page

ivan-kishko opened the pull request:
## Description of the change

> Redesigned data entry page implementation
 - Redesigned summary panel
 - Redesigned publish data panel
 - Redesigned data entry form 
 - Add Help Page

## Type: Non-breaking refactor 

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #118 

## Checklists

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [x] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

ivan-kishko commented:
Hello all! Have a couple comments
 1. Test failed due to DataEntryTest expects context field to be in form but according to new design they are removed from there. I already asked @lilidworkin in slack about what should I do with this, but any comments about this would be great to hear.
 2. Disabled disaggregation tab name inside form should be darkgrey when hovered (but not active). When active it turns blue, and currently i did not change blue to darkgrey on hover. If it has to be changed, please let me know, I guess it is question for you @hobuobi.
 Thanks!

mxosman commented:
> Hello all! Have a couple comments
> 
> 1. 1 test failed due to DataEntryFormTest expects context fields to be in form but according to new design they are removed from there. I already asked @lilidworkin in slack about what should I do with this, but any comments about this would be great to hear.
> 2. Disabled disaggregation tab name inside form should be darkgrey when hovered (but not active). When active it turns blue, and currently i did not change blue to darkgrey on hover. If it has to be changed, please let me know, I guess it is question for you @hobuobi.
>    Thanks!

Hi Ivan! I haven't done a deep dive into the code yet - but just to answer #1 - that's totally fine. Makes sense since the context is no longer in the data entry form and your solution to remove its assertion from the data entry form test is perfect.

lilidworkin commented:
@ivan-kishko  this looks great!! A few tiny things:

1. I think the behavior you have for hovered/active disabled disaggregation tab names is fine! (i.e. turns dark grey on hover, and blue on active).
2. I did notice that hovering on a _non-disabled disaggregation_ tab name doesn't do anything -- should it turn blue? I think that was the previous behavior.
3. If only one metric is disabled, the text at the bottom of the page is slightly wrong grammatically. Instead of 

> Total Staff is associated with your agency’s operations, but have been disabled. If you believe this is incorrect, go to Metric Configuration to re-enable them.

it should be

> Total Staff is associated with your agency’s operations, but has been disabled. If you believe this is incorrect, go to Metric Configuration to re-enable it.

Question for @hobuobi -- I found myself missing the "Back to Reports Overview" link we used to have on the left side. Do we not need that anymore since we have "Save as Draft" instead?
<img width="346" alt="Screen Shot 2022-11-02 at 5 11 27 PM" src="https://user-images.githubusercontent.com/99899591/199603331-b7c6b11d-05b8-481f-9808-2f1a8670fa66.png">

lilidworkin commented:
Oh, and one other tiny thing: in the help panel, the next next to "Breakdowns" seems to be vertically aligned, but instead can it go onto two lines?

That is, instead of 
<img width="658" alt="Screen Shot 2022-11-02 at 5 12 32 PM" src="https://user-images.githubusercontent.com/99899591/199603545-aca846c9-5138-4512-98c7-7f849ba9604d.png">

Can it look like 
<img width="650" alt="Screen Shot 2022-11-02 at 5 13 28 PM" src="https://user-images.githubusercontent.com/99899591/199603670-efee33c1-a86f-4b22-a687-b55c126f57f0.png">

ivan-kishko commented:
> @ivan-kishko this looks great!! A few tiny things:
> 
> 1. I think the behavior you have for hovered/active disabled disaggregation tab names is fine! (i.e. turns dark grey on hover, and blue on active).
> 2. I did notice that hovering on a _non-disabled disaggregation_ tab name doesn't do anything -- should it turn blue? I think that was the previous behavior.
> 3. If only one metric is disabled, the text at the bottom of the page is slightly wrong grammatically. Instead of
> 
> > Total Staff is associated with your agency’s operations, but have been disabled. If you believe this is incorrect, go to Metric Configuration to re-enable them.
> 
> it should be
> 
> > Total Staff is associated with your agency’s operations, but has been disabled. If you believe this is incorrect, go to Metric Configuration to re-enable it.
> 
> Question for @hobuobi -- I found myself missing the "Back to Reports Overview" link we used to have on the left side. Do we not need that anymore since we have "Save as Draft" instead? <img alt="Screen Shot 2022-11-02 at 5 11 27 PM" width="346" src="https://user-images.githubusercontent.com/99899591/199603331-b7c6b11d-05b8-481f-9808-2f1a8670fa66.png">

@Hello @lilidworkin 
1.Got it.
2. If you look at the video that attached to this task hovering over not active and non-disabled disaggregation indeed does not do anything (or maybe mouse was not exactly on item and this is my mistake then). Anyway some hover effect seems reasonable, going to fix it.
3.Sorry did not notice, gonna fix it as well

About that link - now we have "Save as Draft" button which does the same, also you can see that in figma this link is removed (not on demo video though)

mxosman commented on publisher/src/components/Reports/DataEntryForm.tsx:
```diff
@@ -125,6 +124,7 @@ const DataEntryForm: React.FC<{
 
   const isPublished =
     reportStore.reportOverviews[reportID].status === "PUBLISHED";
+  const navigateState = { settingsMenuOption: menuOptions[2] };
```
Nice!

mxosman commented on publisher/src/components/Forms/NotReportedIcon.tsx:
```diff
@@ -50,6 +52,14 @@ export const NotReportedIconImg = styled.img<{
+
+  ${InputWrapper}:hover & {
+    opacity: 1;
+  }
```
Wow! I don't think I realized you could interpolate like this. Just for my understanding, this is targeting the `TabItem` & `InputWrapper` styled-components which are nested inside of the `NotReportedIconImg` (though, I didn't think `NotReportedIconImg` has any children)?

mxosman replied:
Oh wait - these are the parent elements, right?

mxosman replied:
That makes so much sense - thank you for explaining that! Great to know you can do this!

mxosman replied:
> 

100%!

mxosman replied:
> I like this showing / hiding on hover being declared in css rather than setting react state!

100%!

mxosman commented:
Hi Ivan! This looks fantastic - thank you for swimming in the wild DataEntryForm sea! It's really cool to see these pages evolve and I love a lot of the decisions you made.

Here are a few notes from my end:

1. When hovering over a disabled disaggregation tab, the tooltip's opacity appears too faint.
<img width="445" alt="Screen Shot 2022-11-03 at 11 08 26 AM" src="https://user-images.githubusercontent.com/59492998/199780322-e448f54c-0306-40e5-a118-0e3d28720acb.png">

2. Small note: for some reason it feels like the spacing between paragraphs and the line height in the text in the help modal are a bit vertically overcrowded. I'm not sure if Figma has the exact measurements between these areas if it was written in one text box - but I would recommend adding a bit more line height and spacing between paragraphs:
This PR:
<img width="418" alt="Screen Shot 2022-11-03 at 11 16 25 AM" src="https://user-images.githubusercontent.com/59492998/199781018-d2c7c18e-0ee4-43d6-9a1e-b7a8c192fdd0.png">
Figma: <img width="432" alt="Screen Shot 2022-11-03 at 11 40 05 AM" src="https://user-images.githubusercontent.com/59492998/199781162-e750ba6b-e16f-4dcd-b63f-6ca8ccff71f2.png">

3. When the Need Help modal is open, it feels a bit strange to still be able to interact with the data entry form. My recommendation is to disallow any interaction with the layers below the open modal/background.
<img width="528" alt="Screen Shot 2022-11-03 at 11 33 34 AM" src="https://user-images.githubusercontent.com/59492998/199781736-db7f6ead-2725-4179-a12b-8a9df18db3a9.png">

terryttsai commented:
Going off of Mahmoud's comment on how we should disable interaction with the layers below the Need Help modal, we could also make it so that clicking outside the need help modal closes the modal?

ivan-kishko commented:
> Hi Ivan! This looks fantastic - thank you for swimming in the wild DataEntryForm sea! It's really cool to see these pages evolve and I love a lot of the decisions you made.
> 
> Here are a few notes from my end:
> 
> 1. When hovering over a disabled disaggregation tab, the tooltip's opacity appears too faint.
> 
> <img alt="Screen Shot 2022-11-03 at 11 08 26 AM" width="445" src="https://user-images.githubusercontent.com/59492998/199780322-e448f54c-0306-40e5-a118-0e3d28720acb.png">
> 
> 2. Small note: for some reason it feels like the spacing between paragraphs and the line height in the text in the help modal are a bit vertically overcrowded. I'm not sure if Figma has the exact measurements between these areas if it was written in one text box - but I would recommend adding a bit more line height and spacing between paragraphs:
>    This PR:
> 
> <img alt="Screen Shot 2022-11-03 at 11 16 25 AM" width="418" src="https://user-images.githubusercontent.com/59492998/199781018-d2c7c18e-0ee4-43d6-9a1e-b7a8c192fdd0.png">
> 
> Figma:
> <img alt="Screen Shot 2022-11-03 at 11 40 05 AM" width="432" src="https://user-images.githubusercontent.com/59492998/199781162-e750ba6b-e16f-4dcd-b63f-6ca8ccff71f2.png">
> 
> 3. When the Need Help modal is open, it feels a bit strange to still be able to interact with the data entry form. My recommendation is to disallow any interaction with the layers below the open modal/background.
> 
> <img alt="Screen Shot 2022-11-03 at 11 33 34 AM" width="528" src="https://user-images.githubusercontent.com/59492998/199781736-db7f6ead-2725-4179-a12b-8a9df18db3a9.png">

Hello @mxosman , fixed all you asked to in commit [3c07c0a](https://github.com/Recidiviz/justice-counts/pull/133/commits/3c07c0a2259b9e8a28b56bb094f427256d9c48eb)
Have a few comments though:
1. This hovering - I can't reproduce that, thing is - tooltip has no css properties that would set opacity or something like that, can you tell a bit more what you made to make tooltip look like that.
2. In fact figma has like description of what font-sizes and line-heights are used in this text (can not understand why it look different), actually these are from predefined typography theme for project. Anyway you are right, I added some more spacing and changed line-height of subtitle to make it look prettier
3. I did that, but reading what @terryttsai wrote maybe I will have to change it again, will see.

mxosman commented:
> Hello @mxosman , fixed all you asked to in commit [3c07c0a](https://github.com/Recidiviz/justice-counts/pull/133/commits/3c07c0a2259b9e8a28b56bb094f427256d9c48eb) Have a few comments though:
> 
> 1. This hovering - I can't reproduce that, thing is - tooltip has no css properties that would set opacity or something like that, can you tell a bit more what you made to make tooltip look like that.
> 2. In fact figma has like description of what font-sizes and line-heights are used in this text (can not understand why it look different), actually these are from predefined typography theme for project. Anyway you are right, I added some more spacing and changed line-height of subtitle to make it look prettier
> 3. I did that, but reading what @terryttsai wrote maybe I will have to change it again, will see.

Thank you, Ivan!

1. I figured out that it was happening on the 3rd disaggregation tab. Do you have access to other agencies? If so, if you navigate to the agency list and select `Law Enforcement`, go to the Settings > Metric Configuration > Select Total Arrests metric (since it has 3 disaggregation tabs) and disable the 3rd tab (Gender) - attaching a video of the steps:

https://user-images.githubusercontent.com/59492998/199845230-41ca7e0d-eef2-488f-b48a-416440740fd2.mov

Let me know if you can recreate.

2. OK - I'm glad I'm not the only one - this has stumped me before too - not sure why despite using the same predefined font properties, it looks slightly different. Thank you for taking time to make those adjustments!

4. To clarify, I think Terry meant in addition to disabling interaction with the below form when the modal is open, we can also set the behavior of clicking outside of the open modal to trigger the modal to close.

terryttsai commented:
Besides my suggestions, the code looks good to me, thanks Ivan!

ivan-kishko commented:
> > Hello @mxosman , fixed all you asked to in commit [3c07c0a](https://github.com/Recidiviz/justice-counts/pull/133/commits/3c07c0a2259b9e8a28b56bb094f427256d9c48eb) Have a few comments though:
> > 
> > 1. This hovering - I can't reproduce that, thing is - tooltip has no css properties that would set opacity or something like that, can you tell a bit more what you made to make tooltip look like that.
> > 2. In fact figma has like description of what font-sizes and line-heights are used in this text (can not understand why it look different), actually these are from predefined typography theme for project. Anyway you are right, I added some more spacing and changed line-height of subtitle to make it look prettier
> > 3. I did that, but reading what @terryttsai wrote maybe I will have to change it again, will see.
> 
> Thank you, Ivan!
> 
> 1. I figured out that it was happening on the 3rd disaggregation tab. Do you have access to other agencies? If so, if you navigate to the agency list and select `Law Enforcement` (for this example), go to the Settings > Metric Configuration > Select Total Arrests metric (since it has 3 disaggregation tabs) and disable the 3rd tab (Gender) - attaching a video of the steps:
> 
>  Screen.Recording.2022-11-03.at.5.10.52.PM.mov 
> Let me know if you can recreate. It's on report metrics that have 3 disaggregation tabs and the 3rd one is disabled.
> 
> 2. OK - I'm glad I'm not the only one - this has stumped me before too - not sure why despite using the same predefined font properties, it looks slightly different. Thank you for taking time to make those adjustments!
> 3. To clarify, I think Terry meant in addition to disabling interaction with the below form when the modal is open, we can also set the behavior of clicking outside of the open modal to trigger the modal to close.

Thanks for clarifying on steps to repro the issue with hovering disaggregation @mxosman!
So from what I get - it has nothing to do with order but with fill state of dimensions inside disaggregation. This effect you can see when it is disabled but has some filled inputs with or without errors. Basically in this case icon and tooltip have been given additional css with filter property. I just change it to use opacity instead filter and not on hover everything is ok and not hover state looks the same from user perspective. Although not sure why disabled disaggregation keeps values inside its inputs, but I guess this is a question for another day or task)
Made close help page on click away as well. Check out [84a2438](https://github.com/Recidiviz/justice-counts/pull/133/commits/84a243888f21c51f53edf602d4777edf59962209) and [b00d836](https://github.com/Recidiviz/justice-counts/pull/133/commits/b00d836489ab9b561928fb7a43fbb72b6563fc34) commits to look at result! Thanks!

mxosman commented:
> Thanks for clarifying on steps to repro the issue with hovering disaggregation @mxosman! So from what I get - it has nothing to do with order but with fill state of dimensions inside disaggregation. This effect you can see when it is disabled but has some filled inputs with or without errors. Basically in this case icon and tooltip have been given additional css with filter property. I just change it to use opacity instead filter and now on hover everything is ok and not hover state looks the same from user perspective. Although not sure why disabled disaggregation keeps values inside its inputs, but I guess this is a question for another day or task) Made close help page on click away as well. Check out [84a2438](https://github.com/Recidiviz/justice-counts/pull/133/commits/84a243888f21c51f53edf602d4777edf59962209) and [b00d836](https://github.com/Recidiviz/justice-counts/pull/133/commits/b00d836489ab9b561928fb7a43fbb72b6563fc34) commits to look at result! Thanks!

Thank you so much for all the changes, Ivan! Very strange about the tooltip! For some reason I'm still seeing it when I run this branch with your latest commits (HEAD = b00d836):
<img width="528" alt="Screen Shot 2022-11-04 at 9 00 40 AM" src="https://user-images.githubusercontent.com/59492998/199991397-4d3d5d15-137e-4c9d-ba7b-86b0c0410f1f.png">

This is on Law Enforcement > Total Arrests -- do you see the same or no?

Everything else looks good to me!

mxosman commented:
> > Thanks for clarifying on steps to repro the issue with hovering disaggregation @mxosman! So from what I get - it has nothing to do with order but with fill state of dimensions inside disaggregation. This effect you can see when it is disabled but has some filled inputs with or without errors. Basically in this case icon and tooltip have been given additional css with filter property. I just change it to use opacity instead filter and now on hover everything is ok and not hover state looks the same from user perspective. Although not sure why disabled disaggregation keeps values inside its inputs, but I guess this is a question for another day or task) Made close help page on click away as well. Check out [84a2438](https://github.com/Recidiviz/justice-counts/pull/133/commits/84a243888f21c51f53edf602d4777edf59962209) and [b00d836](https://github.com/Recidiviz/justice-counts/pull/133/commits/b00d836489ab9b561928fb7a43fbb72b6563fc34) commits to look at result! Thanks!
> 
> Thank you so much for all the changes, Ivan! Very strange about the tooltip! For some reason I'm still seeing it when I run this branch with your latest commits (HEAD = [b00d836](https://github.com/Recidiviz/justice-counts/commit/b00d836489ab9b561928fb7a43fbb72b6563fc34)): <img alt="Screen Shot 2022-11-04 at 9 00 40 AM" width="528" src="https://user-images.githubusercontent.com/59492998/199991397-4d3d5d15-137e-4c9d-ba7b-86b0c0410f1f.png">
> 
> This is on Law Enforcement > Total Arrests -- do you see the same or no?
> 
> Everything else looks good to me!

It shows up fine when the disaggregation tab is clicked (open) - but when it is not open the tooltip has this grey'd like the screenshot above.

mxosman commented:
I think I found the culprit - try removing the `opacity` all together in the `DisaggregationHasInputIndicator` styled component.

ivan-kishko commented:
> I think I found the culprit - try removing the `opacity` all together in the `DisaggregationHasInputIndicator` styled component.

It is going to work but then disabled disaggregations which have inputs filled will not look different from ones that has only empty inputs. Can't understand why It does not work for you. 

![Screen Shot 2022-11-04 at 5 27 40 PM](https://user-images.githubusercontent.com/84838278/199999643-c7646b46-20b4-43d3-8689-95b26dfb68bd.png)

ivan-kishko commented:
> > I think I found the culprit - try removing the `opacity` all together in the `DisaggregationHasInputIndicator` styled component.
> 
> It is going to work but then disabled disaggregations which have inputs filled will not look different from ones that has only empty inputs. Can't understand why It does not work for you.
> 
> ![Screen Shot 2022-11-04 at 5 27 40 PM](https://user-images.githubusercontent.com/84838278/199999643-c7646b46-20b4-43d3-8689-95b26dfb68bd.png)

@mxosman I figured out why, need some time to resolve it.

ivan-kishko commented:
@mxosman sorry for that issue was not actually solved. Can you please check [92fa6bb](https://github.com/Recidiviz/justice-counts/pull/133/commits/92fa6bbdb20612c815c4d940b11eff0fee5a3f85) commit if you have time?

mxosman reviewed (approved):
> @mxosman sorry for that issue was not actually solved. Can you please check [ad1697e](https://github.com/Recidiviz/justice-counts/pull/133/commits/ad1697e24fc0ac1aba837ebd3e0e929ac70b62c1) commit if you have time?

No apologies necessary at all - I'm sure a lot of this is due my past-self's decisions. Thank you so much for taking the time on this - I really like your solution! Much appreciated.

lilidworkin reviewed (approved):
Thank you, @ivan-kishko ! This should be good to merge! I hope we'll have a new task for you ASAP!
